controllers/index.js

```diff
@@ -9,13 +15,39 @@ exports.getEndpoints = (req, res, next) => {
 }
 
 exports.getTopics = (req, res, next) => {
-    return retrieveTopics()
+    return retrieveTopics('topics')
     .then(({rows}) => {
         res.status(200).send(rows)
     })
     .catch(next) 
 }
 
+exports.getArticlesById = (req, res, next) => {
+    const {params} = req
+    return retrieveArticlesById(params.id)
+    .then(({rows}) => {
+        res.status(200).send(rows[0])
+    })
+    .catch(next) 
+}
+
+exports.getArticles = (req, res, next) => {
+    return retrieveArticles()
+    .then(({rows}) => {
+        res.status(200).send(rows)
+    })
+    .catch(next) 
+}
+
+exports.getCommentsByArticleId = (req, res, next) => {
+    const {params} = req
+    return Promise.all([retrieveComments(params.id), retrieveArticlesById(params.id)])
```

Comment:
Excellent choice to handle the potential error this way as the response from retrieving the comments isn't a reliable error check anymore.